File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/lookup/RequestServerInfoXml.java
package com.lasthopesoftware.bluewater.client.connection.builder.lookup;

import com.lasthopesoftware.bluewater.client.library.repository.Library;
import com.namehillsoftware.handoff.promises.Promise;

import xmlwise.XmlElement;

public interface RequestServerInfoXml {
	Promise<XmlElement> promiseServerInfoXml(Library library);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/lookup/RequestServerInfoXml.kt
package com.lasthopesoftware.bluewater.client.connection.builder.lookup

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise
import org.jsoup.nodes.Document

interface RequestServerInfoXml {
	fun promiseServerInfoXml(libraryId: LibraryId): Promise<Document?>
}
